src/sentry/runner/commands/backup.py

```diff
@@ -182,14 +202,79 @@ def compare(self, on: InstanceID, left: JSONData, right: JSONData) -> list[Compa
         return []
 
 
+class EmailObfuscatingComparator(JSONScrubbingComparator):
+    """Comparator that compares emails, but then safely truncates them to ensure that they
+    do not leak out in logs, stack traces, etc."""
+
+    def __init__(self, *fields: str):
+        super().__init__(*fields)
+
+    def compare(self, on: InstanceID, left: JSONData, right: JSONData) -> list[ComparatorFinding]:
+        findings = []
+        for f in self.fields:
+            if f not in left["fields"] and f not in right["fields"]:
+                continue
+
+            lv = left["fields"][f]
+            rv = right["fields"][f]
+            if lv != rv:
+                lv = self.truncate([lv] if isinstance(lv, str) else lv)[0]
+                rv = self.truncate([rv] if isinstance(rv, str) else rv)[0]
+                findings.append(
+                    ComparatorFinding(
+                        kind=self.get_kind(),
+                        on=on,
+                        reason=f"""the left `{f}` value ("{lv}") on `{on}` was not equal to the
+                                right value ("{rv}")""",
+                    )
+                )
+        return findings
+
+    def scrub(
+        self,
+        left: JSONData,
+        right: JSONData,
+    ) -> None:
+        super().__scrub__(left, right, self.truncate)
+
+    @staticmethod
+    def truncate(data: list[str]) -> list[str]:
+        truncated = []
+        for d in data:
+            parts = d.split("@")
+            if len(parts) == 2:
+                username = parts[0]
+                domain = parts[1]
+                truncated.append(f"{username[0]}...@...{domain[-6:]}")
```

Comment:
is it worth splitting domain by `.` then taking what is to the right of it? In case the 6 lettered domain is a personal domain or something

Reply:
Discussed offline. Agreed that its not worth shooting for perfect obfuscation.